diff -ur getfem-4.3.orig/contrib/xfem_contact/xfem_dirichlet.cc getfem-4.3/contrib/xfem_contact/xfem_dirichlet.cc
--- getfem-4.3.orig/contrib/xfem_contact/xfem_dirichlet.cc	2014-03-15 11:15:09.000000000 +0000
+++ getfem-4.3/contrib/xfem_contact/xfem_dirichlet.cc	2014-10-22 12:42:03.299941536 +0100
@@ -485,7 +485,7 @@
                            &numflag, &nparts, options, &edgecut, &(part[0]));
 # else
   int ncon = 1, edgecut;
-  int options[METIS_NOPTIONS] = { 0 };
+  idx_t options[METIS_NOPTIONS] = { 0 };
   METIS_SetDefaultOptions(options);
   METIS_PartGraphRecursive(&ne, &ncon, &(xadj[0]), &(adjncy[0]), &(vwgt[0]), 0, 0,
                            &nparts, 0, 0, options, &edgecut, &(part[0]));
@@ -899,7 +899,7 @@
                                &numflag, &nparts, options, &edgecut, &(part[0]));
 # else
       int ncon = 1, edgecut;
-      int options[METIS_NOPTIONS] = { 0 };
+      idx_t options[METIS_NOPTIONS] = { 0 };
       METIS_SetDefaultOptions(options);
       METIS_PartGraphRecursive(&ne, &ncon, &(xadj[0]), &(adjncy[0]), &(vwgt[0]), 0, 0,
                                &nparts, 0, 0, options, &edgecut, &(part[0]));
diff -ur getfem-4.3.orig/contrib/xfem_stab_unilat_contact/xfem_stab_unilat_contact.cc getfem-4.3/contrib/xfem_stab_unilat_contact/xfem_stab_unilat_contact.cc
--- getfem-4.3.orig/contrib/xfem_stab_unilat_contact/xfem_stab_unilat_contact.cc	2014-03-15 11:15:09.000000000 +0000
+++ getfem-4.3/contrib/xfem_stab_unilat_contact/xfem_stab_unilat_contact.cc	2014-10-22 12:42:16.134940591 +0100
@@ -518,7 +518,7 @@
                            &numflag, &nparts, options, &edgecut, &(part[0]));
 # else
   int ncon = 1, edgecut;
-  int options[METIS_NOPTIONS] = { 0 };
+  idx_t options[METIS_NOPTIONS] = { 0 };
   METIS_SetDefaultOptions(options);
   METIS_PartGraphRecursive(&ne, &ncon, &(xadj[0]), &(adjncy[0]), &(vwgt[0]), 0, 0,
                            &nparts, 0, 0, options, &edgecut, &(part[0]));
diff -ur getfem-4.3.orig/interface/src/gf_asm.cc getfem-4.3/interface/src/gf_asm.cc
--- getfem-4.3.orig/interface/src/gf_asm.cc	2014-07-04 10:41:32.000000000 +0100
+++ getfem-4.3/interface/src/gf_asm.cc	2014-10-22 13:21:45.540894212 +0100
@@ -166,7 +166,7 @@
   // assemby patch vector
   const getfem::mesh_fem &mf_P0 = getfem::classical_mesh_fem(mesh, 0);
   bgeot::size_type nbe = mf_P0.nb_dof();
-  int ne = 0;
+  idx_t ne = 0;
   double size_of_crack = 0;
   size_of_crack *= h;
   getfem::modeling_standard_plain_vector Patch_Vector(nbe);
@@ -185,8 +185,8 @@
   // std::cout<<"Path_element_list="<< Patch_element_list << std::endl;
   // std::cout<<"Path_dof_ind="<< Patch_dof_ind << std::endl;
   std::cout << "number of element in patch=" << ne << std::endl;
-  std::vector<int> xadj(ne+1), adjncy, numelt(ne), part(ne);
-  std::vector<int> vwgt(ne), indelt(mesh.convex_index().last_true()+1);
+  std::vector<idx_t> xadj(ne+1), adjncy, numelt(ne), part(ne);
+  std::vector<idx_t> vwgt(ne), indelt(mesh.convex_index().last_true()+1);
   std::vector<double> vwgtt(ne);
   int j = 0, k = 0;
   for (dal::bv_visitor ic(Patch_element_list); !ic.finished(); ++ic, j++) {
@@ -214,7 +214,7 @@
   std::cout << "ratio size beween mesh and coarse mesh= " << ratio_size
             << std::endl;
 
-  int nparts = 1;
+  idx_t nparts = 1;
 #ifdef GETFEM_HAVE_METIS
   nparts = int(size_of_crack/(ratio_size*h));
 # ifdef GETFEM_HAVE_METIS_OLD_API
@@ -231,8 +231,8 @@
   METIS_PartGraphRecursive(&ne, &(xadj[0]), &(adjncy[0]), &(vwgt[0]), &(adjwgt[0]), &wgtflag,
                            &numflag, &nparts, options, &edgecut, &(part[0]));
 # else
-  int ncon = 1, edgecut;
-  int options[METIS_NOPTIONS] = { 0 };
+  idx_t ncon = 1, edgecut;
+  idx_t options[METIS_NOPTIONS] = { 0 };
   METIS_SetDefaultOptions(options);
   METIS_PartGraphRecursive(&ne, &ncon, &(xadj[0]), &(adjncy[0]), &(vwgt[0]), 0, 0,
                            &nparts, 0, 0, options, &edgecut, &(part[0]));
diff -ur getfem-4.3.orig/src/getfem/getfem_model_solvers.h getfem-4.3/src/getfem/getfem_model_solvers.h
--- getfem-4.3.orig/src/getfem/getfem_model_solvers.h	2014-03-14 18:56:46.000000000 +0000
+++ getfem-4.3/src/getfem/getfem_model_solvers.h	2014-10-22 12:42:50.439938063 +0100
@@ -810,7 +810,7 @@
                             &numflag, &nbbl, options, &edgecut, &(npart[0]));
 #else
         int ncon = 1, edgcut;
-        int options[METIS_NOPTIONS] = { 0 };
+        idx_t options[METIS_NOPTIONS] = { 0 };
         METIS_SetDefaultOptions(options);
         METIS_PartGraphKway(&ne, &ncon, &(xadj[0]), &(adjncy[0]), 0, 0, 0, &nbbl,
                             0, 0, options, &edgecut, &(npart[0]));
diff -ur getfem-4.3.orig/src/getfem_mesh.cc getfem-4.3/src/getfem_mesh.cc
--- getfem-4.3.orig/src/getfem_mesh.cc	2014-03-14 18:56:46.000000000 +0000
+++ getfem-4.3/src/getfem_mesh.cc	2014-10-22 12:42:37.741938999 +0100
@@ -171,7 +171,7 @@
                       &numflag, &size, options, &edgecut, &(npart[0]));
 #else
   int ncon = 1, edgecut;
-  int options[METIS_NOPTIONS] = { 0 };
+  idx_t options[METIS_NOPTIONS] = { 0 };
   METIS_SetDefaultOptions(options);
   METIS_PartGraphKway(&ne, &ncon, &(xadj[0]), &(adjncy[0]), 0, 0, 0, &size,
                       0, 0, options, &edgecut, &(npart[0]));
